Core/HttpHandlerProxy.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Net;
+using System.Security.Authentication;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.Exchange.WebServices.Data;
+using Microsoft.Exchange.WebServices.NETStandard.Extensions;
+
+namespace Microsoft.Exchange.WebServices.NETStandard.Core
+{
+    internal class HttpHandlerProxy : DelegatingHandler
+    {
+        class SimpleHttpMessageHandlerProxy : DelegatingHandler
+        {
+            public SimpleHttpMessageHandlerProxy(HttpMessageHandler messageHandler)
+                : base(messageHandler) 
+            {
+            }
+
+            internal Task<HttpResponseMessage> SendImplAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                return base.SendAsync(request, cancellationToken);
+            }
+        }
+
+        private readonly string httpClientName;
+        private readonly HttpMessageHandler httpMessageHandler;
+        private readonly HttpClientHandler httpClientHandler;
+
+        /// <summary>
+        /// Gets or sets the cookie container.
+        /// </summary>
+        /// <value>The cookie container.</value>
+        public CookieContainer CookieContainer { get; set; }

```

Comment:
Seem to be not initialized/not used? I'd cover it with tests

Reply:
it's set to request, not to delegating handler?

Reply:
each request creates HttpHandlerProxy and on each request cookie container is set to HttpHandlerProxy, cookie container is managed by ExchangeService and not by HttpHandlerProxy